core/src/main/java/com/linecorp/armeria/server/management/ManagementService.java

```diff
@@ -64,6 +66,16 @@ public final class ManagementService extends AbstractHttpService {
      * Returns a singleton {@link ManagementService}.
      */
     public static ManagementService of() {
+        AppInfoService.INSTANCE.setAppInfo(null);
+        return INSTANCE;
+    }
+
+    /**
+     * Returns a singleton {@link ManagementService} with specified {@link AppInfo}.
+     */
+    public static ManagementService of(AppInfo appInfo) {
+        requireNonNull(appInfo, "appInfo");
+        AppInfoService.INSTANCE.setAppInfo(appInfo);
```

Comment:
This can be overwritten if there are multiple servers in the same JVM.
Although this can give a slight advantage of not allocating an object, I don't think `ManagementService` will be allocated often anyways.
What do you think of just allocating a new `ManagementService`?

e.g.
```java
    public static ManagementService of() {
        return INSTANCE;
    }

    public static ManagementService of(AppInfo appInfo) {
        requireNonNull(appInfo, "appInfo");
        return new ManagementService(appInfo);
    }
```